core/src/com/unciv/ui/worldscreen/WorldScreen.kt

```diff
@@ -184,9 +184,11 @@ class WorldScreen(val gameInfo: GameInfo, val viewingCiv:CivilizationInfo) : Bas
         // Don't select unit and change selectedCiv when centering as spectator
         if (viewingCiv.isSpectator())
             mapHolder.setCenterPosition(tileToCenterOn, immediately = true, selectUnit = false)
-        else
+        else {
             mapHolder.setCenterPosition(tileToCenterOn, immediately = true, selectUnit = true)
-
+            if (viewingCiv.getNextDueUnit() == bottomUnitTable.selectedUnit)
+                viewingCiv.cycleThroughDueUnits()
```

Comment:
This is for the case the first 'due' unit was already selected after the start of a new turn. Not sure how (or if) this worked (i.e. was not the case) before. However, I decided to reimplement it in a more robust way: now `cycleThroughDueUnits()` accepts a unit you don't want to see as next if it can yield anything else.